de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/inout/impl/HUPackingMaterialDAO.java

```diff
@@ -51,10 +59,48 @@
 	@Override
 	public I_M_HU_PackingMaterial retrivePackingMaterialOfProduct(final I_M_Product product)
 	{
-		return Services.get(IQueryBL.class).createQueryBuilder(I_M_HU_PackingMaterial.class, product)
+		return iQueryBL.createQueryBuilder(I_M_HU_PackingMaterial.class, product)
 				.filter(new EqualsQueryFilter<I_M_HU_PackingMaterial>(I_M_HU_PackingMaterial.COLUMNNAME_M_Product_ID, product.getM_Product_ID()))
 				.create()
 				.setOnlyActiveRecords(true)
 				.firstOnly(I_M_HU_PackingMaterial.class);
 	}
+
+	@Nullable
+	@Override
+	public I_M_HU_PackingMaterial retrievePackingMaterialOrNull(@NonNull final PackageId packageId)
+	{
+		return iQueryBL
+				.createQueryBuilder(I_M_Package_HU.class)
+				.addEqualsFilter(I_M_Package_HU.COLUMNNAME_M_Package_ID, packageId)
+				//
+
+				.andCollect(I_M_HU.COLUMN_M_HU_ID, I_M_HU.class)
+				.andCollectChildren(I_M_HU_Item.COLUMN_M_HU_ID)
+				.andCollect(I_M_HU_PackingMaterial.COLUMN_M_HU_PackingMaterial_ID, I_M_HU_PackingMaterial.class)
+				.create()
+				.first(I_M_HU_PackingMaterial.class);
```

Comment:
I'm do not really understand how everything HU-related works, even with all of teo's explanations. If you have suggestions here, i'm glad to check them out, otherwise i'll just leave it as it is.

Plus: all this `andCollect` and `andCollectChildren` looks like whichcraft to me, which again i don't understand how it works. That's why i pasted the sql query above. Frankly, i would have just wanted a simple `.joinOn` method, and i would've been happy.